### PR TITLE
prefix device-specific distinct IDs with '$device:'

### PR DESCRIPTION
**Pull request by csiden**

### CODE REVIEW

File: src/androidTest/java/com/mixpanel/android/mpmetrics/MixpanelBasicTest.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
 import com.mixpanel.android.util.HttpService;
 import com.mixpanel.android.util.RemoteService;
 
+import org.hamcrest.CoreMatchers;
```
Comment (zihejia) [inline src/androidTest/java/com/mixpanel/android/mpmetrics/MixpanelBasicTest.java:17]: nit: redundant

File: src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java [hunk 2]
```diff
@@ -1779,7 +1777,7 @@ public void identify(String distinctId) {
                 MPLog.e(LOGTAG, "Can't identify with null distinct_id.");
                 return;
             }
-            if (distinctId != mPersistentIdentity.getEventsDistinctId()) {
+            if (!distinctId.equals(mPersistentIdentity.getEventsDistinctId())) {
```
Comment (zihejia) [inline src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java:1780]: good catch!

File: README.md [hunk 3]
```diff
@@ -158,7 +158,7 @@ We'd also love for you to come and work with us! Check out our **[opening positi
 <a name="changelog"></a>
 # Changelog
 
-See [wiki page](https://github.com/mixpanel/mixpanel-android/wiki/Changelog).
+See [wiki page](CHANGELOG.md).
```
Comment (zihejia) [inline README.md:161]:
```suggestion
See [changelog](CHANGELOG.md).
```

**Review by zihejia — APPROVED** [review]
LGTM!

**Comment by carstenhag** [comment]
Hey there, we just noticed that on AND, some distinct IDs have a "$device:" prefix. On iOS this is never the case. We are confused as to why this is the case? :D

Screenshot 2023-07-28 at 10 25 22

**Comment by jaredmixpanel** [comment]
@carstenhag mixpanel-swift started prefixing `$device:` in v4.0.5: https://github.com/mixpanel/mixpanel-swift/releases/tag/v4.0.5

The Objective-C SDK started w/ v5.0.2: https://github.com/mixpanel/mixpanel-iphone/releases/tag/v5.0.2

Both were released 5 months ago. If you are using a version >= those, then I would recommend reaching out to support@mixpanel.com for help troubleshooting any identity management issues you're experiencing.

**Pull request merged**